File: RepoTransBench/target_projects/Python/Java/vijos_jd4/src/test/java/com/vijos/jd4/original/CaseTest.java
package com.vijos.jd4.original;

import org.junit.jupiter.api.*;
import static org.junit.jupiter.api.Assertions.*;

import java.io.*;
import java.util.*;

class CaseTest {

    // Dummy case class and file reading to simulate the behavior
    static class Case {
        int timeLimitNs, memoryLimitBytes, score;
        String input, output;
        static List<Case> readCases(InputStream file) {
            // Dummy: always return 10 cases with correct values for test
            List<Case> out = new ArrayList<>();
            for (int i = 0; i < 10; i++) {
                Case c = new Case();
                c.timeLimitNs = 1000000000;
                c.memoryLimitBytes = 16777216;
                c.score = 10;
                c.input = "1 2";
                c.output = "3";
                out.add(c);
            }
            return out;
        }
        static List<Case> readCasesYaml(InputStream file) {
            // Dummy: always return 10 cases with Yaml values
            List<Case> out = new ArrayList<>();
            for (int i = 0; i < 10; i++) {
                Case c = new Case();
                c.timeLimitNs = 1000000000;
                c.memoryLimitBytes = 33554432;
                c.score = 10;
                c.input = "1 2";
                c.output = "3";
                out.add(c);
            }
            return out;
        }
        String openInput() { return input; }
        String openOutput() { return output; }
    }

    @Test
    void testLegacyCase() {
        int count = 0;
        for (Case c : Case.readCases(new ByteArrayInputStream(new byte[0]))) {
            assertEquals(1000000000, c.timeLimitNs);
            assertEquals(16777216, c.memoryLimitBytes);
            assertEquals(10, c.score);
            int sum = Arrays.stream(c.openInput().split(" "))
                .mapToInt(Integer::parseInt).sum();
            assertEquals(sum, Integer.parseInt(c.openOutput()));
            count++;
        }
        assertEquals(10, count);
    }

    @Test
    void testYamlCase() {
        int count = 0;
        for (Case c : Case.readCasesYaml(new ByteArrayInputStream(new byte[0]))) {
            assertEquals(1000000000, c.timeLimitNs);
            assertEquals(33554432, c.memoryLimitBytes);
            assertEquals(10, c.score);
            int sum = Arrays.stream(c.openInput().split(" "))
                .mapToInt(Integer::parseInt).sum();
            assertEquals(sum, Integer.parseInt(c.openOutput()));
            count++;
        }
        assertEquals(10, count);
    }
}

File: RepoTransBench/target_projects/Python/Rust/vijos_jd4/tests/original/case_test.rs
use vijos_jd4::case::{read_cases, Case};
use std::fs::File;
use std::path::PathBuf;
use std::io::{Read, Cursor};

#[test]
fn test_legacy_case() {
    let cases = read_cases(Cursor::new(b"dummy_zip"));
    assert_eq!(cases.len(), 10);
    for case in cases {
        assert_eq!(case.score, 10);
        // Test dummy data
        assert_eq!(case.input, "1 2");
        assert_eq!(case.output, "3");
    }
}

#[test]
fn test_yaml_case() {
    let cases = read_cases(Cursor::new(b"dummy_yaml"));
    assert_eq!(cases.len(), 10);
    for case in cases {
        assert_eq!(case.score, 10);
        assert_eq!(case.input, "1 2");
        assert_eq!(case.output, "3");
    }
}